### PR TITLE
Add example for optional attr

### PR DESCRIPTION
**Pull request by Rageking8**
## Summary

Add an example for usage of `OptionalAttribute`.

### CODE REVIEW

File: xml/System.Runtime.InteropServices/OptionalAttribute.xml [hunk 1]
```diff
@@ -62,6 +62,15 @@
  You can apply this attribute to parameters.  
   
  Optional parameters are not supported by all languages.  
+
+ <xref:System.Runtime.InteropServices.OptionalAttribute> is sometimes used in conjunction with <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute>.
+  
+
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/OptionalAttribute.xml:68]:
```suggestion
```

File: xml/System.Runtime.InteropServices/OptionalAttribute.xml [hunk 2]
```diff
@@ -62,6 +62,15 @@
  You can apply this attribute to parameters.  
   
  Optional parameters are not supported by all languages.  
+
+ <xref:System.Runtime.InteropServices.OptionalAttribute> is sometimes used in conjunction with <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute>.
+  
+   
+
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/OptionalAttribute.xml:69]:
```suggestion
```

File: xml/System.Runtime.InteropServices/OptionalAttribute.xml [hunk 3]
```diff
@@ -62,6 +62,15 @@
  You can apply this attribute to parameters.  
   
  Optional parameters are not supported by all languages.  
+
+ <xref:System.Runtime.InteropServices.OptionalAttribute> is sometimes used in conjunction with <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute>.
+  
+   
+  
+## Examples
+ The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.OptionalAttribute> attribute to a parameter of a method written in C#.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/OptionalAttribute.xml:71]:
```suggestion

The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.OptionalAttribute> attribute to a parameter of a method written in C#.
```

File: xml/System.Runtime.InteropServices/OptionalAttribute.xml [hunk 4]
```diff
@@ -62,6 +62,15 @@
...
  Optional parameters are not supported by all languages.  
+
+ <xref:System.Runtime.InteropServices.OptionalAttribute> is sometimes used in conjunction with <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute>.
+  
+   
+  
+## Examples
+ The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.OptionalAttribute> attribute to a parameter of a method written in C#.  
+  
+ :::code language="csharp" source="~/snippets/csharp/System.Runtime.InteropServices/OptionalAttribute/Overview/sample.cs" id="Snippet1":::
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/OptionalAttribute.xml:73]:
```suggestion
:::code language="csharp" source="~/snippets/csharp/System.Runtime.InteropServices/OptionalAttribute/Overview/sample.cs" id="Snippet1":::
```

File: xml/System.Runtime.InteropServices/OptionalAttribute.xml [hunk 5]
```diff
@@ -62,6 +62,15 @@
  You can apply this attribute to parameters.  
   
  Optional parameters are not supported by all languages.  
+
+ <xref:System.Runtime.InteropServices.OptionalAttribute> is sometimes used in conjunction with <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute>.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/OptionalAttribute.xml:66]:
```suggestion
xref:System.Runtime.InteropServices.OptionalAttribute> is sometimes used in conjunction with <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute>.
```

File: xml/System.Runtime.InteropServices/OptionalAttribute.xml [hunk 6]
```diff
@@ -62,6 +62,15 @@
...
+
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/OptionalAttribute.xml:67]:
```suggestion

```

**Review by gewarren — APPROVED** [review]
Thanks @Rageking8. Just some whitespace cleanup suggestions (the weird indentation and trailing spaces are a relic from long ago).

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by Rageking8** [comment]
@gewarren Thanks for the quick review, applied all the space cleanups suggested!

**Comment by Rageking8** [comment]
@jkoritzinsky Gentle ping for review.

**Pull request merged**